### Kursbezeichnung:

Projekt: Programmierung mit Python (DLBFTPPP01)

### Aufgabenstellung 2:

Vergleich von Aktien

### Studiengang:

Softwareentwicklung

### Datum:

xx.xx.2026

### Verfasser:

Dustin Vetter, Matrikelnummer IU14110854

# Wer gewinnt das KI-Wettrüsten?

## Ein Aktienvergleich entlang der KI-Wertschöpfungskette

### Gliederung

1. Einleitung
2. Projektvorgehen
3. Kennzahlen zu Rendite und Risiko
4. Gesamtbild und Vergleich
5. Fazit und Reflexion

# 1. Einleitung

## Aktualität und Relevanz des Themas

- Veröffentlichung von ChatGPT Ende 2022 als öffentlich wahrgenommener Auslöser des aktuellen KI-Booms
- Seitdem deutlich steigende Investitionen großer Technologiekonzerne in KI-Infrastruktur
- Generative KI-Anwendungen haben sich zu einem strategischen Schwerpunkt der gesamten Technologiebranche entwickelt

## Untersuchungsgegenstand

- Der öffentlichen Diskurs dreht sich hauptsächlich um generative KI, z. B. ChatGPT, Claude und Mistral AI
- Diese Unternehmen sind privat finanziert und (noch) nicht börsennotiert, es liegen keine Kursdaten vor
- Deshalb werden zentrale Investitionspartner betrachtet mit unmittelbaren Bezug zur KI-Wertschöpfungskette

### Untersuchte Hardware-Unternehmen

- ASML (Niederlande): Lithografieanlagen für die Chipfertigung
- TSMC (Taiwan): größter Auftragsfertiger von Halbleitern
- NVIDIA (USA): marktführende KI-Grafikprozessoren

### Untersuchte Software-Unternehmen

- Microsoft (USA): Cloud-Plattform Azure, Investor bei OpenAI (ChatGPT)
- Amazon (USA): Cloud-Plattform AWS, Investor bei Anthropic (Claude)
- Google (USA): eigene KI-Modelle und KI-Chips

## Zielsetzung und Leitfrage

### Leitfrage

Bewegen sich die Aktienkurse von Unternehmen derselben Ebene (Hardware bzw. Software) stärker im Gleichlauf als die Kurse von Unternehmen unterschiedlicher Ebenen?

### Annahme

* Aktien innerhalb derselben Ebene vehalten sich tendenziell ähnlicher als über die Ebenen hinweg
* Hardware-Ebene: stärkerer Gleichlauf zu erwarten, da sch die Unternehmen in der Lieferkette ergänzen statt konkurrieren
* Software-Ebene: schwächerer Gleichlauf zu erwarten, da die Unternehmen im direkten Wettbewerb um Marktanteile stehen

### Überprüfung

Die Annahme wird im weiteren Verlauf anhand einer Korrelationsanalyse der Aktienrendite überprüft. Das methodische Vorgehen dazu folgt im nächsten Kapitel.

# 2. Projektvorgehen

## Import der benötigten Bibliotheken

In [1]:
import yfinance
import pandas
import plotly.express

- yfinance für den Datenabruf
- pandas für die Datenverarveitung
- plotly für die interaktive Visualisierung

## Abruf der Kursdaten

In [2]:
companies = [
    "NVDA", "TSM", "ASML", "MSFT", "AMZN", "GOOGL"
]

data = yfinance.download(companies, start="2021-08-01", end="2026-07-31", auto_adjust=True)

[*********************100%***********************]  6 of 6 completed


* **start**: Startdatum, 01.08.2021
* **end**: Enddatum, 31.07.2026
* **auto_adjust**: Bereinigt Kurse um Aktiensplits und Dividenden

### Begriffserklärung

- **Aktiensplit**: Aufteilung einer Aktie in mehrere neue Aktien bei gleichbleibendem Gesamtwert des Investments (TODO: Quelle)
    - Beispiel: 1 Aktie zu 1000 USD wird zu 10 Aktien zu je 100 USD
    - Ohne Bereinigung würde ein Split in den historischen Daten wie ein Kurssturz aussehen
* **Dividende**: Ausschüttung eines Teils des Unternehmensgewinns an die Aktionäre (TODO: Quelle)
    - Der Kurs sinkt am Auszahlungstag rechnerisch um den ausgeschütteten Betrag, ohne dass ein realer Wertverlust vorliegt

## Prüfen, ob die Daten korrekt geladen wurden

In [3]:
data.head()

Price            Close                                                 \
Ticker            AMZN        ASML       GOOGL        MSFT       NVDA   
Date                                                                    
2021-08-02  166.574005  734.308472  133.669220  273.340759  19.681923   
2021-08-03  168.311996  743.734863  134.437912  275.548126  19.746698   
2021-08-04  167.735992  758.952454  133.937866  274.962677  20.204115   
2021-08-05  168.799500  758.255981  135.053955  277.851410  20.565865   
2021-08-06  167.246994  747.312866  134.545471  277.793823  20.295797   

Price                         High                                      ...  \
Ticker             TSM        AMZN        ASML       GOOGL        MSFT  ...   
Date                                                                    ...   
2021-08-02  107.721443  167.945999  740.214235  134.146978  275.212150  ...   
2021-08-03  109.150246  169.550003  744.612602  134.539005  275.653708  ...   
2021-08-04  109.454437  169.444000  761.900590  134.457263  275.999136  ...   
2021-08-05  109.896912  169.449997  768.645939  135.151600  277.956992  ...   
2021-08-06  108.965897  168.750000  751.043299  135.273029  277.832219  ...   

Price             Open                                       Volume          \
Ticker           GOOGL        MSFT       NVDA         TSM      AMZN    ASML   
Date                                                                          
2021-08-02  133.929410  274.818671  19.632096  108.311394  67078000  499900   
2021-08-03  133.947252  273.916659  19.671956  108.136261  83146000  533700   
2021-08-04  134.159896  274.684357  19.921093  109.168682  43678000  629700   
2021-08-05  134.487975  275.317826  20.429338  109.758640  48670000  723700   
2021-08-06  134.846316  276.882129  20.453254  109.113388  52752000  556900   

Price                                               
Ticker         GOOGL      MSFT       NVDA      TSM  
Date                                                
2021-08-02  21634000  16267400  217444000  6028800  
2021-08-03  20762000  17879000  301811000  5874700  
2021-08-04  19676000  16191300  231309000  4254000  
2021-08-05  17870000  13900200  211435000  4452300  
2021-08-06  20488000  16589300  178497000  4338100  

[5 rows x 30 columns]

### Begriffserklärung
* **Close**: Eröffnungskurs, erster gehandleter Preis am jeweiligen Handelstag
* **High**: Höchster gehandelter Preis am jeweiligen Handelstag
* **Low**: Niedrigstrer gehandleter Preis am jeweiligen Handelstag
* **Open**: Schlusskurs, letzter gehandelter Preis am jeweiligen Handelstag
* **Volume**: Handelsvolumen, Anzahl der an diesem Tag gehandelten Aktien

## Nur die Schlusskurse anzeigen

* Der Schlusskurs (**Close**) ist für die Rendite- und Risikoberechnung relevante Referenzgröße

In [4]:
close = data["Close"]
close.head()

Ticker,AMZN,ASML,GOOGL,MSFT,NVDA,TSM
Date,,,,,,
2021-08-02,166.574005,734.308472,133.669220,273.340759,19.681923,107.721443
2021-08-03,168.311996,743.734863,134.437912,275.548126,19.746698,109.150246
2021-08-04,167.735992,758.952454,133.937866,274.962677,20.204115,109.454437
2021-08-05,168.799500,758.255981,135.053955,277.851410,20.565865,109.896912
2021-08-06,167.246994,747.312866,134.545471,277.793823,20.295797,108.965897


## Bedeutung der Zahlen

* **166.574005**: Eine Aktie von Amazon hatte am Schlusskurs vom 02.08.2021 einen Wert von ca. 166,57 USD

## Tägliche Rendite berechnen

### Begriffserklärung

- **Rendite**: prozentuale Wertveränderung einer Kapitalanlage über einen bestimmten Zeitraum, bezogen auf das eingesetzte Kapitel (TODO: Quelle)
    - Beispiel: Ein Investment von 100 EUR wächst auf 110 EUR => Rendite = 10 %
- **Tägliche Rendite**: Rendite bezogen auf einen einzelnen Handelstag (TODO: Quelle)
    - Formel: `(Kurs heute - Kurs Vortag) / Kurs Vortag`
- Grundlage für alle weiteren Kennzahlen

In [6]:
returns = close.pct_change().dropna()
returns.head()

Ticker,AMZN,ASML,GOOGL,MSFT,NVDA,TSM
Date,,,,,,
2021-08-03,0.010434,0.012837,0.005751,0.008076,0.003291,0.013264
2021-08-04,-0.003422,0.020461,-0.003720,-0.002125,0.023164,0.002787
2021-08-05,0.006340,-0.000918,0.008333,0.010506,0.017905,0.004043
2021-08-06,-0.009197,-0.014432,-0.003765,-0.000207,-0.013132,-0.008472
2021-08-09,-0.000918,0.006894,0.008652,-0.003904,-0.003486,0.000085


### Bedeutung der Zahlen

- **0.010434**: Tägliche Rendite von Amazon am 03.08.2021
    - Der Kurs ist gegenüber dem Vortag (02.08.2021) um 1,0434 % gestiegen
- **-0.003422**: Tägliche Rendite von Amazon am 04.08.2021
    -  Der Kurs ist gegenüber dem Vortag (03.08.2021) um 0,3422 % gefallen